prow/cmd/marque/main.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+const (
+	// Time before generating the first cert, for safety.
+	sleepTime = time.Minute
+	// Time between renewals.
+	renewTime = 12 * time.Hour
+)
+
+func main() {
+	kc, err := kube.NewClientInCluster("default")
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting kube client.")
+	}
+
+	root, err := ioutil.TempDir("", "certbot")
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not create temp dir.")
+	}
+	http.Handle("/", http.FileServer(http.Dir(root)))
+	go http.ListenAndServe(":http", nil)
+
+	logrus.Infof("Sleeping for %v before generating cert.", sleepTime)
+	time.Sleep(sleepTime)
+	if err := generate(root); err != nil {
+		logrus.WithError(err).Fatal("Error getting cert.")
+	}
+	if err := replaceSecret(kc); err != nil {
+		logrus.WithError(err).Fatal("Error updating secrets.")
+	}
+
+	for range time.Tick(renewTime) {
+		logrus.Info("Renewing.")
+		if err := renew(); err != nil {
+			logrus.WithError(err).Warning("Error renewing cert.")
+		}
+		if err := replaceSecret(kc); err != nil {
+			logrus.WithError(err).Warning("Error updating secrets.")
+		}
+	}
+}
+
+func generate(root string) error {
+	args := []string{
+		"certonly",
+		"--webroot",
+		"-w", root,
+		"-d", "prow.k8s.io",
```

Comment:
That's going to be a bother, since GCE ingress doesn't support SNI. I'll do it in a future PR, because I have to handle TLS manually to get it to work, rather than use the ingress object. In that PR I'll also make it redirect http to https.